sqlglot/parser.py

```diff
@@ -2754,6 +2739,20 @@ def _parse_table(
             exp.Expression, bracket or self._parse_bracket(self._parse_table_parts(schema=schema))
         )
 
+        if self._match_texts(("AT", "BEFORE")):
```

Comment:
what happens if at or before are aliases?

Reply:
oof good point

Reply:
One option is to try to match the time travel syntax and if we don't succeed we can safely assume that the `AT` or `BEFORE` is an alias and retreat. What do you think about this one? Snowflake allows these keywords as aliases, e.g. this works:

```sql
with foo as (select 1 as c) select at.c from foo at
```

Reply:
what if you match at ( ? text seq

Reply:
also is this snowflake only?

Reply:
yea this before / at syntax seems snowflake only, maybe I can move it to its `_parse_table_parts` instead..

> what if you match at ( ? text seq

It's a bit annoying because you can have `with foo as (select 1 as c) select at.c from foo at(c)` for example, so the `(` in its own is not enough. I think the FARROW token would be enough.

Reply:
yea maybe movve it to snowflake somehow. why can’t you to match textseq (at, ()??

Reply:
ok addressed